integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextmove/servicebus/ServiceBusAmqpClient.java

```diff
@@ -0,0 +1,105 @@
+                properties.getNextmove().getServiceBus().getSasKeyName(),
+                serviceBusUtil.getSasKey());
+
+        ServiceBusProcessorClient serviceBusProcessorClient = new ServiceBusClientBuilder()
+                .connectionString(connectionString)
+                .retryOptions(new AmqpRetryOptions()
+                        .setDelay(Duration.ofSeconds(3))
+                        .setMaxRetries(Integer.MAX_VALUE))
+                .processor()
+                .queueName(serviceBusUtil.getLocalQueuePath())
+                .maxConcurrentCalls(50)
+                .receiveMode(ServiceBusReceiveMode.PEEK_LOCK)
+                .disableAutoComplete()
+                .processMessage(this::processMessage)
+                .processError(this::processError)
+                .buildProcessorClient();
+        serviceBusProcessorClient.start();
+    }
+
+    private void processMessage(ServiceBusReceivedMessageContext context) {
+        ServiceBusReceivedMessage m = context.getMessage();
+        log.debug(format("Received message on queue=%s with id=%s", serviceBusUtil.getLocalQueuePath(), m.getMessageId()));
+        ServiceBusPayload payload;
+        try {
+            payload = payloadConverter.convert(m.getBody().toBytes());
+        } catch (IOException e) {
+            log.error(String.format("Failed to convert servicebus message with id = %s, abandoning", m.getMessageId()), e);
+            context.deadLetter();
+            return;
+        }
+        InputStream asicStream = (payload.getAsic() != null) ? new ByteArrayInputStream(Base64.getDecoder().decode(payload.getAsic())) : null;
+        nextMoveQueue.enqueueIncomingMessage(payload.getSbd(), DPE, asicStream);
+        context.complete();
+    }
+
+    private void processError(ServiceBusErrorContext context) {
+        log.error("Error when receiving messages from namespace: '{}'. Entity: '{}'",
+                context.getFullyQualifiedNamespace(), context.getEntityPath());
+
+        if (!(context.getException() instanceof ServiceBusException)) {
+            log.error("Non-ServiceBusException occurred", context.getException());
+            return;
+        }
+
+        ServiceBusException exception = (ServiceBusException) context.getException();
+        ServiceBusFailureReason reason = exception.getReason();
+
+        if (reason == ServiceBusFailureReason.MESSAGING_ENTITY_DISABLED
+                || reason == ServiceBusFailureReason.MESSAGING_ENTITY_NOT_FOUND
+                || reason == ServiceBusFailureReason.UNAUTHORIZED) {
+            log.error("An unrecoverable error occurred. Stopping processing with reason {}: {}", reason, exception.getMessage());
+        } else if (reason == ServiceBusFailureReason.MESSAGE_LOCK_LOST) {
+            log.error("Message lock lost for message", context.getException());
+        } else if (reason == ServiceBusFailureReason.SERVICE_BUSY) {
+            try {
+                TimeUnit.SECONDS.sleep(1);
```

Comment:
bør ein retrye fleire gonger om den framleis er opptatt etter 1 sec ?

Reply:
ServiceBusProcessorClient er satt opp med retry. Dette er bare  for at den skal vente litt (ihht. anbefalinger).